build.rs

```diff
@@ -0,0 +1,245 @@
+            }
+            if let Some(clash) = used_networks.insert(account_type.name(), account_type.clone()) {
+                return Err(format!(
+                    "networks must be unique but this account's network:\n{:#?}\nclashed with\n:{:#?}",
+                    account_type,
+                    clash
+                ));
+            }
+            if account_type.network.is_empty() {
+                return Err("network is mandatory.".into());
+            }
+
+            if let Err(err) = is_valid_rust_identifier(&account_type.name()) {
+                return Err(format!(
+                    "network not valid: {} for {:#?}",
+                    err, account_type
+                ));
+            }
+            if account_type.decimals.len() != account_type.symbols.len() {
+                return Err(format!(
+                    "decimals must be specified for each symbol: {:?}",
+                    account_type
+                ));
+            }
+            if let Some(sig_type) = &account_type.standard_account {
+                match sig_type.as_str() {
+                    "Sr25519"| "Ed25519" | "secp256k1" | "*25519" => {},
+                    _ => {
+                        return Err(format!("Unknown sig type in standardAccount: expected one of Sr25519, Ed25519, secp256k1, *25519: {:?}", account_type))
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+struct AccountType {
+    prefix: u16,
+    network: String,
+    display_name: String,
+    /// If standard account is None then the network is reserved.
+    standard_account: Option<String>,
+    symbols: Vec<String>,
+    decimals: Vec<u8>,
+    website: Option<String>,
+}
+
+impl AccountType {
+    fn name(&self) -> String {
+        format!(
+            "{}Account",
+            inflector::cases::pascalcase::to_pascal_case(&self.network)
+        )
+    }
+
+    fn is_reserved(&self) -> bool {
+        self.standard_account.is_none()
+    }
+}
+
+fn consecutive_runs(data: &[u16]) -> (Vec<u16>, Vec<u16>) {
+    let mut slice_start = 0_u16;
+    let (mut starts, mut ends) = (Vec::new(), Vec::new());
+    for i in 1..data.len() {
+        if data[i - 1] + 1 != data[i] {
+            starts.push(slice_start);
+            ends.push(data[(i - 1)]);
+            slice_start = data[i];
+        }
+    }
+    if !data.is_empty() {
+        starts.push(slice_start);
+        ends.push(data[data.len() - 1]);
+    }
+    (starts, ends)
+}
+
+fn create_ss58_registry(json: &str) -> Result<proc_macro2::TokenStream, String> {
+    let registry: Registry =
+        serde_json::from_str(json).map_err(|e| format!("json parsing error: {}", e))?;
+    registry.is_valid()?;
+
+    let mut registry = registry.registry;
+
+    let mut ordered_prefixes = registry.iter().map(|i| i.prefix).collect::<Vec<_>>();
+    ordered_prefixes.sort_unstable();
```

Comment:
Don't use sort_unstable

Reply:
It's sorting primitives - there's no difference between 6 and 6 in this case - it's just a tad faster. In general I agree we don't want non-determinism but here I believe it makes no difference which we use.

Reply:
Please use `sort()`.

> it's just a tad faster.

We are sorting here like in maximum 256 elements, no one will be able to tell the difference.